File: spark/learning-spark-lightning-fast-big-data-analysis/jvm/src/main/scala/redocpot/spark/FlatMap.scala
package scala.redocpot.spark

import org.apache.spark.SparkConf
import org.apache.spark.SparkContext
import org.apache.spark.SparkContext._

object FlatMap {
  def main(args : Array[String]) : Unit = {
    val conf = new SparkConf().setAppName("FlatMap").setMaster("local[*]")
    val sc = new SparkContext(conf)
    val lines = sc.parallelize(List("hello world", "hi"))
    val words = lines.flatMap(line => line.split(" "))
    println(words.collect)
    sc.stop
  }
}

File: spark/learning-spark-lightning-fast-big-data-analysis/jvm/src/main/java/redocpot/spark/FlatMap.java
package redocpot.spark;

import java.util.Arrays;
import java.util.Iterator;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.FlatMapFunction;

public class FlatMap {
    public static void main(String[] args) {
        SparkConf conf = new SparkConf()
            .setMaster("local")
            .setAppName("FlatMap");
        JavaSparkContext sc = new JavaSparkContext(conf);

        JavaRDD<String> lines = sc.parallelize(Arrays.asList("hello world", "hi"));
        JavaRDD<String> words = lines.flatMap(new FlatMapFunction<String, String>() {
            public Iterator<String> call(String line) {
                return (Iterator<String>) Arrays.asList(line.split(" "));
            }
        });
        System.out.println(words.collect());

        sc.stop();
    }
}
